go-controller/pkg/util/nicstobridge.go

```diff
@@ -331,3 +332,39 @@ func BridgeToNic(bridge string) error {
 	klog.Infof("Successfully deleted OVS bridge %q", bridge)
 	return nil
 }
+
+// GetSmartNICHostInterface returns the host representor interface attached to bridge
+func GetSmartNICHostInterface(bridgeName string) (string, error) {
+	stdout, stderr, err := RunOVSVsctl("list-ports", bridgeName)
+	if err != nil {
+		return "", fmt.Errorf("failed to get list of ports on bridge %q:, stderr: %q, error: %v",
+			bridgeName, stderr, err)
+	}
+
+	for _, port := range strings.Split(stdout, "\n") {
+		stdout, stderr, err = RunOVSVsctl("get", "Port", port, "Interfaces")
+		if err != nil {
+			return "", fmt.Errorf("failed to get port %q on bridge %q:, stderr: %q, error: %v",
+				port, bridgeName, stderr, err)
+
+		}
+		// remove brackets on list of interfaces
+		ifaces := strings.TrimPrefix(strings.TrimSuffix(stdout, "]"), "[")
+		for _, iface := range strings.Split(ifaces, ",") {
+			stdout, stderr, err = RunOVSVsctl("get", "Interface", strings.TrimSpace(iface), "Name")
+			if err != nil {
+				return "", fmt.Errorf("failed to get Interface %q Name on bridge %q:, stderr: %q, error: %v",
+					iface, bridgeName, stderr, err)
+
+			}
+			flavor, err := GetSriovnetOps().GetRepresentorPortFlavour(stdout)
```

Comment:
Do we want to consider two PFs from the same SmartNIC in this PR? The flows on SmartNIC physical bridge may also need change.

Reply:
I am not sure I fully understand the use-case with having two PFs on the same bridge.
Do you mean VF LAG ? if so, to my understanding, its using a linux bond and the bond will be connected to the bridge.

I think VF LAG or other physical bridge configuration scenarios should be addressed in a separate PR.

Reply:
> I am not sure I fully understand the use-case with having two PFs on the same bridge.
> Do you mean VF LAG ? if so, to my understanding, its using a linux bond and the bond will be connected to the bridge.

Yes, VF LAG, or the ability to use both PF reps. If we do bond on Host, how the PF rep on SmartNIC would need to be plugged in OVS?
> 
> I think VF LAG or other physical bridge configuration scenarios should be discussed and addressed in a separate PR.
Yes, this won't be the immediate use case to address.

Reply:
okay, good to know how bond works on SmartNIC, I will give it a try. Thanks!